codec.go

```diff
@@ -222,8 +222,11 @@ func (cc *LengthFieldBasedFrameCodec) Encode(c Conn, buf []byte) (out []byte, er
 type innerBuffer []byte
 
 func (in *innerBuffer) readN(n int) (buf []byte, err error) {
-	if n <= 0 {
-		return nil, errors.New("zero or negative length is invalid")
+	if n == 0 {
+		return []byte{}, nil
```

Comment:
It can be `return nil, nil` directly, no need to initialize an empty []byte, standard Go functions accept nil buffers, they can be used just like []byte{}.